Dockerfile

```diff
@@ -99,7 +99,7 @@ RUN KEY="A4A9406876FCBD3C456770C88C718D3B5072E1F5" \
     && gpgconf --kill all \
     rm -rf "${GNUPGHOME}"; \
     apt-key list > /dev/null \
-    && echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-5.6" | tee -a /etc/apt/sources.list.d/mysql.list \
+    && echo "deb http://repo.mysql.com/apt/debian/ stretch mysql-5.7" | tee -a /etc/apt/sources.list.d/mysql.list \
```

Comment:
Instead of installing this debian package to be able to use libmysqlclient-dev, you could have also installed mariadb drivers (libmariadb-dev mariadb-client) and do not need installing any deb packages.
I have no preference here but I am just curious why have you decided to go with this solution?

Reply:
We had a long discussion on that some time ago (i might try to find it If you want). There were different opinions - MySQL vs. MariaDB, both have pros and cons. I think no-one had strong opinion on that, but I believe in corporate world MySQL is a bit more "popular" choice. Especially when corporates are already cooperating with Oracle. I actually had a PR that was trying to switch to Maria DB some time ago. 

Also - related - we are using those client libs/database to run tests of MySQL Hook as well. From what I remember there is at least one test case we have and functionality in MySQL Hook (with importing data via LOAD_DATA/LOAD_DATA_LOCAL) that will not work with MariaDB (i tried it looooong time ago) - with importing raw data (it is deemed insecure by MariaDB and removed -  it failed with mariadb client even if the server was MySQL and supported it). https://dev.mysql.com/doc/refman/5.7/en/load-data-local.html

Ideally we should test both - but with the current Travis queue capacity we cannot have full matrix of tests (this might change when we finally switch to Github Actions).

Also there is a related PR in progress (https://github.com/apache/airflow/pull/65760). It is about MySQL Hook rather than Airflow's Metastore, but we might take a look and implement more flexibility/more various combinations.

Reply:
I just wanted to know the reasoning. It makes total sense. Thanks for the answer.